### PR TITLE
Cdrw 756 consent part1

### PR DESCRIPTION
**Pull request by splig**

### CODE REVIEW

File: src/components/AspspSelect.vue [hunk 1]
```diff
@@ -6,33 +6,34 @@
...
-  computed: {
-    name() {
-      return this.aspsp.name;
+  export default {
+    name: 'aspsp-select',
+    props: ['aspsp'],
+    computed: {
+      name() {
+        return this.aspsp.name;
+      },
```
Comment (robmckinnon) [inline src/components/AspspSelect.vue]: The Vue.js convention is to not indent after the <script> and <style> tags in a Single File Component. For example see the docs here: https://vuejs.org/v2/guide/single-file-components.html

Can we remove the extra indent in <script> and <style> for consistency?

File: src/store/request.js [hunk 2]
```diff
@@ -28,6 +29,22 @@ const options = (aspsp) => {
   };
 };
 
+const asyncAwaitPostJson = async (endpoint, aspsp, data, unauthorizedType) => {
+  const { headers } = makeHeaders(aspsp);
+  headers['Content-Type'] = 'application/json';
+  const response = await fetch(accountRequestConsentUri, {
```
Comment (robmckinnon) [inline src/store/request.js]: Just spotted this should be `fetch(endpoint`.

Then we can remove `accountRequestConsentUri` from this file.

File: src/store/actions.js [hunk 3]
```diff
@@ -17,6 +17,11 @@ const actions = {
     }
     return commit(types.LOGIN_ERROR);
   },
+  accountRequestAuthoriseConsent(options, data) {
+    const aspspId = data.orgId;
+    const body = { authorisationServerId: data.id };
+    postJson('/account-request-authorise-consent', aspspId, body, types.LOGOUT);
```
Comment (robmckinnon) [inline src/store/actions.js]: The url should be this one:

`const accountRequestConsentUri = (process.env.ACCOUNT_REQUEST_CONSENT_URI || 'http://localhost:8003/account-request-authorise-consent');`

File: src/store/actions.js [hunk 4]
```diff
@@ -17,6 +17,11 @@ const actions = {
     }
     return commit(types.LOGIN_ERROR);
   },
+  accountRequestAuthoriseConsent(options, data) {
+    const aspspId = data.orgId;
+    const body = { authorisationServerId: data.id };
+    postJson(accountRequestConsentUri, aspspId, body, types.LOGOUT);
```
Comment (daniel-kuras) [inline src/store/actions.js]: @splig the response from /account-request-authorise-consent' will be redirect 302 or error. 
Error is the easy one - system may try to retry or take down redirection page and tell user something went wrong. 
302 is more problematic as the UI app needs to redirect the whole page to ASASP AS server but can it be done automatically by Vue/fetch or we need to manually handle the promise from fetch and redirect manually using router.go(path) for Vue.... I don't know

File: src/store/request.js [hunk 5]
```diff
@@ -28,6 +29,22 @@ const options = (aspsp) => {
...
 
+const asyncAwaitPostJson = async (endpoint, aspsp, data, unauthorizedType) => {
+  const { headers } = makeHeaders(aspsp);
+  headers['Content-Type'] = 'application/json';
+  const response = await fetch(endpoint, {
+    method: 'POST',
+    headers,
+    body: JSON.stringify(data),
+  });
+  if (response.status === 204) {
```
Comment (daniel-kuras) [inline src/store/request.js]: As my comment above we need use case for 302 as /account-request-authorise-consent will never return 2XX - only [3-5]XX

**Pull request merged**